pkg/services/searchV2/index.go

```diff
@@ -846,30 +846,96 @@ func (l sqlDashboardLoader) loadAllDashboards(ctx context.Context, limit int, or
 			dashboardQuerySpan.SetAttributes("dashboardUID", dashboardUID, attribute.Key("dashboardUID").String(dashboardUID))
 			dashboardQuerySpan.SetAttributes("lastID", lastID, attribute.Key("lastID").Int64(lastID))
 
-			rows := make([]*dashboardQueryResult, 0)
+			var slices [][]string
 			err := l.sql.WithDbSession(dashboardQueryCtx, func(sess *db.Session) error {
-				sess.Table("dashboard").
-					Where("org_id = ?", orgID)
+				sql := "select id, uid, is_folder, folder_id, slug, data, created, updated from dashboard where org_id = ?"
+				sqlAndArgs := []interface{}{"", orgID}
```

Comment:
Is this a potential maintainability tradeoff for performance by not using the helper function and explicitly naming fields from dashboard table (i.e. fields can be modified over time / removed potentially?)

If I am understanding this correctly though we are reducing number of times we query DB to be just once vs. several times throughout function which seems reasonable

Reply:
Yeah, this is going the raw route and doing the `raw data from db` -> `simple object` conversion ourselves rather than leaving it alone to `xorm`. Unfortunately the automatic conversion is slow (second screenshot from https://github.com/grafana/grafana/pull/56933) so this saves us a bunch of time. The query pattern is exactly the same, its still one query per batch; the changes are about what happens once we receive the data back from the database

> explicitly naming fields from dashboard table (i.e. fields can be modified over time / removed potentially?)

We were explicitly naming the fields from the dashboard table in the code, see https://github.com/grafana/grafana/blob/3a4ad8756e961fbc08db7234e7b973bc9e4fbf48/pkg/services/searchV2/index.go#L1008-L1010. The maintainability still goes down though - its more code thats not super obvious and has random positional digits sprinkled in it 😬